### PR TITLE
Added subcommand for Streams cog "list"

### PR DESCRIPTION
**Pull request by The-Tasty-Jaffa**
This is a temporary thing (before v3) to allow for users to list all the current stream names in the server the command was called in.

### CODE REVIEW

File: cogs/streams.py [hunk 1]
```diff
@@ -291,11 +290,33 @@ def __init__(self, bot):
         if ctx.invoked_subcommand is None:
             await self.bot.send_cmd_help(ctx)
 
+    @streamset.command(pass_context=True, name = "list")
+    @checks.admin()
+    async def List_streams(self, ctx):
+        """Returns all the names of streams for current server""" #Should also add the channel names (for easy of use... just maybe)
+        streams = (self.twitch_streams, self.hitbox_streams, self.mixer_streams, self.picarto_streams)
```
Comment (tekulvw) [inline cogs/streams.py]: `streams = self.twitch_streams + self.hitbox_streams + self.mixer_streams + self.picarto_streams`

File: cogs/streams.py [hunk 2]
```diff
@@ -291,11 +290,33 @@ def __init__(self, bot):
...
+        stream_list = []
+        
+        
+        for channel in set(ctx.message.server.channels):#..
```
Comment (tekulvw) [inline cogs/streams.py]: This whole block should just get changed. With the above changes in place:

```py
in_chan_streams = filter(lambda s: ctx.message.channel.id in s["CHANNELS"], streams)
stream_names = [s["NAME"] for s in in_chan_streams]
```

edit (for in server streams):
```py
def stream_in_channels(stream, channels):
    return any(c.id in stream["CHANNELS"] for c in channels)
in_server_streams = filter(lambda s: stream_in_channels(s, ctx.message.server.channels), streams)
stream_names = [s["NAME"] for s in in_server_streams]
```

**Review by tekulvw — CHANGES_REQUESTED** [review]

**Review by The-Tasty-Jaffa — COMMENTED** [review]
Updated with suggested changes

**Comment by RSNFreud** [comment]
1 month and no update?

**Comment by Kowlin** [comment]
Red v2 (on which this PR is based) is feature locked. I don't know how we will approach old v2 PR's, but please hold on while we get trough the work load in the following weeks.

**Comment by tekulvw** [comment]
Gonna say no because of the feature lock, feel free to comment on/contribute to #782

**Pull request closed without merging**